### ISSUE TITLE
Question about the implementation of "encoder-normalize-before" in transformer.py

### ISSUE THREAD

**Issue by LividWo** [pytorch/fairseq#1938]
#### What is your question?
In the implementation of transformer.py, there is a parameter called "encoder-normalize-before", which is explained as "apply layernorm before each encoder block".

However, when I read the code, it seems that it is applied **after** all encoder layers? I attached the code snippet from transformer.py line 469-481 below, which is taken from the TransformerEncoder class.

 It would be great if you have time to help explain it a bit for me. Thanks a lot.

#### Code

```python
for layer in self.layers:
    # add LayerDrop (see https://arxiv.org/abs/1909.11556 for description)
    dropout_probability = torch.empty(1).uniform_()
    if not self.training or (dropout_probability > self.encoder_layerdrop):
        x = layer(x, encoder_padding_mask)
        if return_all_hiddens:
            assert encoder_states is not None
            encoder_states.append(x)

if self.layer_norm is not None:
    x = self.layer_norm(x)
    if return_all_hiddens:
        encoder_states[-1] = x      
```

#### What have you tried?
I searched the docs and the issues, but found nothing about this fine-grain detail.
#### What's your environment?

 - fairseq Version (0.9.0)

**Comment by YonghyunRyu** [comment]
This name has to be changed.

**Comment by usefulhyun** [comment]
This works here 'https://github.com/facebookresearch/fairseq/blob/main/fairseq/modules/transformer_layer.py'.
No problem.
